### ISSUE TITLE
 Lifecycle docs comments - 20160930 

### ISSUE THREAD

**Issue by thisisaaronland** [whosonfirst/whosonfirst-cookbook#3]
https://github.com/whosonfirst/whosonfirst-cookbook/commit/56f128ce3470fc29dee3abc1bfd5145fe4af7bb1
## 

tl;dr – On the question of assigning new parent IDs, "significant events" and supersedes events: Absolutely not and I will exercise editorial prerogative over this.
## 

The long version follows:

```
Once a feature is given a wof:id, that feature will maintain that wof:id for it's entire life cycle, unless that feature experiences a Significant Event.
```

This doesn't make any sense or rather is confusing absent a discussion about notions of place and any given place having multiple representations across time. This is akin to the idea, in library sciences, that there is a single "work" and multiple "editions" of that work.

It probably makes more sense to speak of records rather than features because whether or not a given record experiences a significant event it still maintains its wof:id

```
Changing a feature's wof:name without storing the original wof:name as an alt-name
```

Alternative name

```
Changing a feature's wof:name due to the original wof:name being wrong to being with
```

... to begin with

```
Giving a feature a new wof:parent_id when the parent's wof:placetype is a country or region
```

No no no. No. See also the "Mike's house in Oakland" example. I could spend all day coming up with examples why limiting it to country or region is a bad idea. Yes, it makes things difficult (at least from a data processing point of view). That is the work.

I can not stress this enough: Absolutely not and I will exercise editorial prerogative over this.

```
The feature's raw data (geojson)...
```

GeoJSON

```
The duplicate (new) feature...
```

The newly duplicated

```
This history is not inherent to the linked feature list, but rather the linked feature list and GitHub log.
```

The linked list feature? The GitHub log? These are both being mentioned for the first time, without context. Also it is important to distinguish between an idealized audit trail that does not exist today and the use of the Git log as the closest practical approximation in 2016.

```
birth, life, death
```

I am still not crazy about this nomenclature. WOF strives for generic and neutral (to the point of being boring) terms wherever possible. Perhaps, CREATE, ALTER and DEPRECATE instead?

"Death" is especially confusing. The record in question does not suddenly cease to exist. The WOF ID will _always_ be honoured. That is the promise of WOF so to say that it "dies" suggests that it will vanish from WOF entirely.

A more "accurate" term given using the model you've adopted would be to say those records which experience "death" are actually "still-born" but that only serves to highlight the awkwardness of the terminology.

```
If the new feature will not have any descendants
```

If the new feature does not have any descendants...

```
However, if the new feature parents any existing Who's On First records, this feature will need to be placed in the hierarchy of all of its descendants.
```

It is important to indicate that in many cases the newly created record will also trigger superseded versions of those descendants.

```
When an existing feature in Who's On First ceases to exist in the real-world ... it may be replaced with a new feature that takes it's place.
```

Again, won't someone think of the children?

```
Descendant's wof:hierarchy before import of parent:
```

Don't make the reader have to try and figure out that newly added element to the hierarchy is the county. Call it out. Where did this new county come from? Is it replacing anything? Is it data that should have been there before? As in: Is the new county simply an update to reflect a reality that existed before the WOF record was created? It is not clear.

```
If an error correction needed to occur to move Iceland, for example,
```

What about Australia? Like really. It's apparently moving around right now.
## the diagram

```
Were more than 10 attributes edited?
```

Wait, what? Where did this come from? Is there some sort of ranking of which 10 fields get updated? Does this include newly added attributes? 

This is not a good rationale for triggering a supersedes event. Things get corrected, or simply updated, all the time in manners that don't substantially change the nature of a place (which is the purpose and the function of a supersedes event).

Further, this is not discussed anywhere in the text.

```
Change to a wof:parent
```

See notes above. Again, no.

```
Change to a wof:placetype
```

What does whether or not the original wof:name being incorrect have to do with this?
## 

cc @nvkelso @dphiffer

**Comment by nvkelso** [comment]
I hear your concerns. Let's talk more about this Monday, please.

**Comment by stepps00** [comment]
The most recent comment ([here](https://github.com/whosonfirst/whosonfirst-cookbook/commit/705faa16ab85376c4d70185ff052e5f9eeeff4cd)) updated the lifecycle document and flowchart. 

I will update if there are missed issues or concerns, otherwise I'll close this issue tomorrow.

**Comment by stepps00** [comment]
Closing this issue - all comments (including new comments from this morning) should be addressed in the most current version of the life cycle document.
